### ISSUE TITLE
Add platform specific flags to CMake build

### ISSUE THREAD

**Issue by kgerheiser** [NOAA-EMC/WW3#630]
**Is your feature request related to a problem? Please describe.**

The makefile build sets specific optimization flags for different machines in https://github.com/NOAA-EMC/WW3/blob/develop/model/bin/cmplr.env

But the CMake build doesn't do this.

**Describe the solution you'd like**

Add a mechanism to set flags for particular machines. My idea is to handle it like UFS here: https://github.com/ufs-community/ufs-weather-model/blob/34a5895b847f276335bd40f03321011d05d1b328/CMakeLists.txt#L68-L77

The env variable `CMAKE_Platform` is used to set these flags. Values would be like `hera.intel`. CMake would detect this value and include a .cmake file which sets compiler flags.

**Comment by ukmo-ccbunney** [comment]
@kgerheiser If I read that correctly, we would have a `configure_<platofrm>.cmake` in the `cmake` directory for each of the system/compiler combinations in `cmpl.env` ? Is that correct?

That looks sensible to me, but I guess the only thing I would say is that a few years ago the `cmplr.env` file was generated to avoid having lots of individual system/compiler `comp` and `link` scripts in the `model/bin` directory (it was getting very cluttered!) This approach looks like it would go back to that; not that this is a necessarily a bad thing, but I wonder if is possible to reproduce something link the cmplr.env script functionality in CMAKE? Just a thought...

@aliabdolali , @mickaelaccensi , @JessicaMeixner-NOAA , @thesser1  - What do others think about this?

**Comment by JessicaMeixner-NOAA** [comment]
Good point @ukmo-ccbunney 

I think ideally we prefer to have one file that has if statements for the different compilers if possible.  This is also because many platforms need the same flags and it makes things significantly simpler for updating and maintaining.  Would it be possible to use the same concept of a CMAKE_platform but only have 1 file to maintain instead of many @kgerheiser, similar to what we have in cmpl.env?

**Comment by kgerheiser** [comment]
Sure, that works too.

**Comment by aliabdolali** [comment]
I agree with the unification of all configure_<platofrm>.cmake files. Thanks, @ukmo-ccbunney for raising, and thanks @kgerheiser for working on it.
